core/templates/paged_array.h

```diff
@@ -278,27 +278,25 @@ class PagedArray {
 
 		count -= remainder;
 
-		uint32_t src_pages = p_array._get_pages_in_use();
+		uint32_t src_page_index = 0;
 		uint32_t page_size = page_size_mask + 1;
 
-		for (uint32_t i = 0; i < src_pages; i++) {
+		while (p_array.count > 0) {
 			uint32_t page_count = _get_pages_in_use();
 			uint32_t new_page_count = page_count + 1;
 
 			if (unlikely(new_page_count > max_pages_used)) {
 				_grow_page_array(); //keep out of inline
 			}
 
-			page_data[page_count] = p_array.page_data[i];
-			page_ids[page_count] = p_array.page_ids[i];
-			if (i == src_pages - 1) {
```

Comment:
For the records, the issue was here. The last page was assumed to be partially filled. If it happened to contain as many items as the page size (page full), `count` would be incremented by zero instead of by the page size.

This line inside this branch would prove so: `DEV_ASSERT((p_array.count & page_size_mask) != 0);`.